### PR TITLE
Remove dependency from Rhino; enable Custom Representation Mesh; adds batch upload to send larger models

### LINKED ISSUE #80: Remove dependency on Rhino mesh representation to allow loading into Excel and Dynamo
#### Description:

As in the issue description.

#### Expected behaviour:

Remove dependency on Rhino

### LINKED ISSUE #85: Allow custom representation mesh to be used
#### Description:

If a "CustomRepresentation" is found (for now, in the CustomData), then use it instead of calculating a MeshRepresentation for the object.

This enables any mesh to be used as a representation for the object, for example the ones pulled from Revit: https://github.com/BHoM/Revit_Toolkit/pull/686

### LINKED ISSUE #87: Mid-large size models can't be pushed
#### Description:

Due to `SpeckleCore`'s timeout limitation. `SpeckleCore`'s 
```cs
SpeckleClient.StreamUpdateAsync(SpeckleStreamId, SpeckleStream)
```
does not actually perform a batch upload like the Grasshopper SpeckleSender does.

After chat with Dimitrie, this is by design, each SpeckleUI does its batching system.

Need to implement some batching in the Speckle_Toolkit Push.
Dimitrie suggest copying over this: https://github.com/speckleworks/SpeckleRhino/blob/master/SpeckleGrasshopper/BaseComponents/GhSenderCoreClient.cs#L784-L891

@rwemay 

#### Steps to reproduce:

Send something like a bit over 90 `Bars`.
You get a timeout error.

#### Expected behaviour:

All bars should be pushed. Size limit should be much more.

### PR DESCRIPTION
**Pull request by alelom**
### NOTE: Depends on 

https://github.com/BHoM/BHoM/pull/839
https://github.com/BHoM/TriangleNet_Toolkit/issues/27
   
### Issues addressed by this PR

Closes #80
Closes #85 
Closes #87 

### Test files

In the toolkit: `\Speckle_Toolkit\Test Scripts\#86-RemoveRhinoDep-CustomMesh`

### Changelog

- Dependency on Rhino removed
- You can now send large models (implemented batch upload as explained in #87 )
- If you put a RenderMesh in CustomData, that is used as a representation.

### CODE REVIEW

**Comment by alelom** [comment]
@al-fisher I'd need CI to add TriangleNet_Toolkit in the dependencies.
Although this PR is limited to the customMesh (so no RenderMesh reference is in place yet), I added some references to methods in TriangleNet_Toolkit to avoid code duplication.

**Comment by FraserGreenroyd** [comment]
@alelom I have added `TriangleNet_Toolkit` as a dependency of `Speckle_Toolkit`.

Can you also remember to set the `assignee`, `labels`, and `project` (should be `SCRUM pull request tracker`) for this and all PRs within the series of work this work is solving :smile:

**Comment by alelom** [comment]
@al-fisher 
Code works now as expected:

- BHoM Representations are constructed **without dependency on Rhino**:
   - First, we check if there is a** custom representation (`RenderMesh`) in the object's Customdata**. If there is, we simply convert it into a `SpeckleMesh`.
   - If not, computing of `GeometricalRepresentation()` is attempted.
   - If a geometrical representation exists, we attempt to translate that in Speckle Geometries (letting speckle do the meshing).
   - If the previous fails, compute the `RenderMesh` out of the Geometrical Representation.
   - If all has failed, return a `SpeckleAbstract` object.

image

**Comment by alelom** [comment]
@rwemay @al-fisher this solves all the main issues we had.

- Dependency on Rhino removed
- You can now (finally!) send large models (I implemented batch upload as explained in #87 )
- If you put a RenderMesh in CustomData, that is used.

File: Speckle_Adapter/Speckle_Adapter.csproj [hunk 1]
```diff
@@ -85,17 +93,17 @@
...
-    <Reference Include="RhinoCommon, Version=5.1.30000.16, Culture=neutral, PublicKeyToken=552281e97c755530, processorArchitecture=MSIL">
-      <HintPath>..\packages\RhinoCommon.5.12.50810.13095\lib\net35\RhinoCommon.dll</HintPath>
+    <Reference Include="Serialiser_Engine">
+      <HintPath>..\..\BHoM_Engine\Build\Serialiser_Engine.dll</HintPath>
       <Private>False</Private>
     </Reference>
     <Reference Include="SpeckleCore, Version=1.7.3.0, Culture=neutral, processorArchitecture=MSIL">
       <HintPath>..\packages\SpeckleCore.1.7.3.417\lib\netstandard2.0\SpeckleCore.dll</HintPath>
-      <Private>False</Private>
+      <Private>True</Private>
```
Comment (al-fisher) [inline Speckle_Adapter/Speckle_Adapter.csproj:106]:
```suggestion
      <Private>False</Private>
```
I think we agreed this should be `false`. Along with SpeckleCoreGeometryClasses.
Sorry @alelom - can you double check. Thanks

Comment (alelom) [reply Speckle_Adapter/Speckle_Adapter.csproj]: Sorry - I got confused on the commit. Updated now.

**Pull request merged**